var/spack/repos/builtin/packages/upcxx/package.py

```diff
@@ -76,7 +102,72 @@ def setup_dependent_build_environment(self, env, dependent_spec):
             env.set('UPCXX_NETWORK', 'aries')
 
     def install(self, spec, prefix):
-        env['CC'] = self.compiler.cc
-        env['CXX'] = self.compiler.cxx
-        installsh = Executable("./install")
-        installsh(prefix)
+        # UPC++ follows autoconf naming convention for LDLIBS, which is 'LIBS'
+        if (env.get('LDLIBS')):
+            env['LIBS'] = env['LDLIBS']
+
+        if spec.version <= Version('2019.9.0'):
+            env['CC'] = self.compiler.cc
+            if '+mpi' in self.spec:
+                if 'platform=cray' in self.spec:
+                    env['GASNET_CONFIGURE_ARGS'] += \
+                        " --with-mpicc=" + self.compiler.cc
+                else:
+                    env['CXX'] = spec['mpi'].mpicxx
+            else:
+                env['CXX'] = self.compiler.cxx
+            installsh = Executable("./install")
+            installsh(prefix)
+        else:
+            if 'platform=cray' in self.spec:
+                # Spack loads the cray-libsci module incorrectly on ALCF theta,
+                # breaking the Cray compiler wrappers
+                # cray-libsci is irrelevant to our build, so disable it
+                for var in ['PE_PKGCONFIG_PRODUCTS', 'PE_PKGCONFIG_LIBS']:
+                    env[var] = ":".join(
+                        filter(lambda x: "libsci" not in x.lower(),
+                               env[var].split(":")))
+                # Undo spack compiler wrappers:
+                # the C/C++ compilers must work post-install
+                # hack above no longer works after the fix to UPC++ issue #287
+                real_cc = join_path(env['CRAYPE_DIR'], 'bin', 'cc')
+                real_cxx = join_path(env['CRAYPE_DIR'], 'bin', 'CC')
+                # workaround a bug in the UPC++ installer: (issue #346)
+                env['GASNET_CONFIGURE_ARGS'] += \
+                    " --with-cc=" + real_cc + " --with-cxx=" + real_cxx
+                if '+mpi' in self.spec:
+                    env['GASNET_CONFIGURE_ARGS'] += " --with-mpicc=" + real_cc
+            else:
+                real_cc = self.compiler.cc
+                real_cxx = self.compiler.cxx
+                if '+mpi' in self.spec:
+                    real_cxx = spec['mpi'].mpicxx
+
+            env['CC'] = real_cc
+            env['CXX'] = real_cxx
+
+            installsh = Executable("./configure")
+            installsh('--prefix=' + prefix)
+
+            make()
+
+            make('install')
+
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    def test_install(self):
+        if self.spec.version <= Version('2019.9.0'):
```

Comment:
you can also `@when` this out at the function level if you want.

Reply:
I like having the warning message below to clarify the test stage was skipped for older releases, even if it only appears on the console when `-v` is passed (and in the build log). I could not find a documented way to force non-verbose console output for such status messages.